src/libs/cmd_lib.py

```diff
@@ -0,0 +1,111 @@
+"""
+Generic definition of a command, possibly including a reference to a renderer.
+
+TODO: In reality, this should be a completely separate library in a completely
+separate repo. But we'll export it later as needed. Again, it's not strictly
+needed that this library exists serverside, since everything should just be
+passed over to the server as a JSON by an agent, Python or otherwise...
+but then you'd need to implement something similar to the below anyways.
+
+This library must not import any other internal libraries; it may only import
+the standard library and external packages.
+"""
+
+import abc
+
+from enum import Enum
+from pydantic import BaseModel
+
+
+class RendererType(str, Enum):
+    """
+    Enumeration of available renderer types.
+    """
+
+    TABULAR = "tabular"
+    FILE_LISTING = "file_listing"
+
+
+class RendererBase(BaseModel, abc.ABC):
+    """
+    Abstract definition of a command response renderer.
+
+    This class provides functionality for interpreting the results of specific
+    instances of CommandBase (i.e. specific commands); it should return a valid
+
+    FIXME: Note that Mythic has specific render "types", such as generic tabular
+    data or file listings; currently, renders provide (un)styled HTML directly.
+    In the future, this may instead be consistent with Mythic's approach instead.
+    """
+
+    @property
+    @abc.abstractmethod
+    def renderer_type(self) -> RendererType:
+        """
+        The specific "format" of the return data.
+        """
+        pass
+
+    @abc.abstractmethod
+    def render_response(self, data: bytes) -> str:
+        """
+        Render a response as valid HTML.
+
+        In the future, this may change to simply being a JSON document, which
+        does not require a change to this specific interface; it would be up to
+        the web interface to deal with the change in presentation.
+        """
+        pass
+
+
+class CommandBase(BaseModel, abc.ABC):

```

Comment:
this is missing the interface for specifying available arguments and requires a `CommandArgBase` class, to be implemented in a different feature branch